src/nhs_grid.jl

```diff
@@ -337,6 +337,41 @@ function update_grid!(neighborhood_search::GridNeighborhoodSearch{<:Any, Paralle
     return neighborhood_search
 end
 
+@inline function foreach_neighbor(f, system_coords, neighbor_system_coords,
+                                  neighborhood_search::GridNeighborhoodSearch, point;
+                                  search_radius = search_radius(neighborhood_search))
```

Comment:
so this is now the default for `GridNeighborhoodSearch`? 
This means that `eachneighbor` is only called in the generic `foreach_neighbor` loop?

If this is the faster approach, why isn't it the default for all `foreach_neighbor` loops?

Reply:
This is the "searching in neighboring cells" part unrolled. It only works for grid-based neighborhood searches.
I only kept `eachneighbor` for backwards-compatibility. I will rework the API in the future (it's needed for interpolation around a position where there is no particle).